apis/v1alpha1/nginxproxy_types.go

```diff
@@ -114,3 +120,69 @@ type TelemetryExporter struct {
 	// +kubebuilder:validation:Pattern=`^(?:http?:\/\/)?[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?(?:\.[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)*(?::\d{1,5})?$`
 	Endpoint string `json:"endpoint"`
 }
+
+// RewriteClientIP specifies the configuration for rewriting the client's IP address.
+type RewriteClientIP struct {
+	// Mode defines how NGINX will rewrite the client's IP address.
+	// There are two possible modes:
+	// - ProxyProtocol: NGINX will rewrite the client's IP using the PROXY protocol header.
+	// - XForwardedFor: NGINX will rewrite the client's IP using the X-Forwarded-For header.
+	// Sets NGINX directive real_ip_header: https://nginx.org/en/docs/http/ngx_http_realip_module.html#real_ip_header
+	//
+	// +optional
+	Mode *RewriteClientIPModeType `json:"mode,omitempty"`
+
+	// SetIPRecursively configures whether recursive search is used when selecting the client's address from.
+	// the X-Forwarded-For header. It is used in conjunction with TrustedAddresses.
```

Comment:
I don't think this is specific to `X-Forwarded-For`, it's for whatever is set in `real_ip_header` or in our case, `Mode`.

Reply:
It is specific to `X-Forwarded-For` - https://github.com/nginxinc/nginx-gateway-fabric/blob/main/docs/proposals/rewrite-client-ip.md#real_ip_recursive